src/screens/PlrStaking/PlrStakingValidator.tsx

```diff
@@ -0,0 +1,617 @@
+
+    const stakeTransactions = await buildStakingTransactions(stkPlrAmount, plrToken);
+
+    if (!stakeTransactions?.transactions || !!stakeTransactions.errorMessage) {
+      return;
+    }
+
+    return { transactions, stakeTransactions: stakeTransactions.transactions };
+  };
+
+  const getFeeInfo = async (transactions: any[]) => {
+    setCalculatingGas(true);
+    const estimate = await estimateTransactions(transactions, gasFeeAsset);
+    const feeInfo = await buildEtherspotTxFeeInfo(estimate.estimation);
+    if (feeInfo?.fee) setTxFeeInfo(feeInfo.fee);
+    else setTxFeeInfo(null);
+    setCalculatingGas(false);
+
+    return feeInfo;
+  };
+
+  const submitStakingTransactions = async () => {
+    if (!stakeTxPayload) return;
+
+    setProcessing(false);
+    const tx = await sendTransactions(selectedChain, stakeTxPayload);
+
+    if (tx) {
+      const stakedToken = await getStakedToken();
+      const stkPlrBalance = await getBalanceForAddress(CHAIN.ETHEREUM, stakedToken);
+      setStartingStkPlr(stkPlrBalance);
+      setIsStaking(true);
+    }
+  };
+
+  const submitTransactions = async () => {
+    const { transactions, stakeTransactions } = await buildStakeTransactions();
+
+    if (!stakeTransactions?.length) return;
+
+    setProcessing(true);
+    setIsBridging(false);
+    setStartingPlr(plrBalance);
+    setLockedStakingAmount(stkPlrAmount);
+
+    let transactionPayload: TransactionPayload = null;
+    let transactionType;
+
+    if (isBridgeTransaction) {
+      // Separate bridge offer txs from staking txs
+      setIsBridging(true);
+
+      transactionType = TRANSACTION_TYPE.EXCHANGE;
+      transactionPayload = mapTransactionsToTransactionPayload(selectedChain, transactions);
+      transactionPayload = {
+        ...transactionPayload,
+        offer: {
+          ...offerData,
+          swapFeeInfo,
+        },
+        type: transactionType,
+        gasToken: gasFeeAsset,
+      };
+
+      let stakeTransactionPayload = mapTransactionsToTransactionPayload(CHAIN.ETHEREUM, stakeTransactions);
+      stakeTransactionPayload = {
+        ...stakeTransactionPayload,
+        offer: null,
+        type: TRANSACTION_TYPE.STAKEPLR,
+      };
+
+      setStakeTxPayload(stakeTransactionPayload);
+    } else {
+      try {
+        transactionType = TRANSACTION_TYPE.STAKEPLR;
+        const allTransactions = [...transactions, ...stakeTransactions];
+        transactionPayload = mapTransactionsToTransactionPayload(CHAIN.ETHEREUM, allTransactions);
+        transactionPayload = {
+          ...transactionPayload,
+          offer: null,
+          type: transactionType,
+        };
+      } catch (e) {
+        console.log('create staking payload error', e);
+      }
+    }
+
+    if (!transactionPayload) return;
+
+    const tx = await sendTransactions(selectedChain, transactionPayload);
+
+    if (!tx) {
+      showTransactionRevertedToast();
```

Comment:
Can we display something more useful? This message drives everyone mad 😂

Reply:
Haha yeah, I was unsure what else I should put here, as the reason it's erroring would be unknown to us - we just get "transaction reverted" in the logs

Reply:
Of course 😂 how about "Sorry, an error occured whilst trying to perform this action. Please try again."